### ISSUE TITLE
Incorrect domain scaling for Neumann BCS in Chebop2.

### ISSUE THREAD

**Issue by nickhale** [chebfun/chebfun#1561]
Chebop2 doesn't seem to be correctly scaling derivatives in boundary conditions when the domain is changed from [-1 1 -1 1] to something else.

Here's an example:
```
N = chebop2(@(x, y, u) laplacian(u), [0 1 0 1]);
N.ubc = 0; N.dbc = 0; N.lbc = 0;
N.rbc = @(x, u) diff(u) - sin(2*pi*x);
u = N\0;
dudx = diffx(u);
dudx1 = dudx(1,:);
plot(dudx1), shg
norm(dudx1, inf)
```
```
ans =
    1.999999999999984
```
I specified `dudx = sin(2*pi*x)` at `x = 1`, it appears I'm getting `dudx = 2*sin(2*pi*x)`. Changing the domain to `[.5 1 .5 1]` I get 
```
norm(dudx1, inf)
ans =
    4.000000000002359
```

**Comment by ajt60gaibb** [comment]
Thanks. This is because the AD information is not being passed through `constructBC.m` correctly.

**Comment by asgeirbirkis** [comment]
Closed by #1565 .
